lib/index.js

```diff
@@ -182,37 +182,39 @@ Mixpanel.prototype.track = function(track, fn){
 
 Mixpanel.prototype.screen =
 Mixpanel.prototype.page = function(msg, fn){
-  var batch = new Batch;
   var category = msg.category();
-  var name = msg.fullName();
+  var name = msg.name();
+  var fullName = msg.fullName(); // includes both category and name
   var self = this;
 
   // all
   if (this.settings.trackAllPages) {
-    batch.push(track(msg.track()));
+    send(msg.track(), fn);
+    return;
   }
 
   // categorized
-  if (category && this.settings.trackCategorizedPages) {
-    batch.push(track(msg.track(category)));
+  if (this.settings.trackCategorizedPages && category) {
+    if (name) {
+      send(msg.track(fullName), fn);
+      return;
+    }
+    send(msg.track(category), fn);
+    return;
   }
 
   // named
-  if (name && this.settings.trackNamedPages) {
-    batch.push(track(msg.track(name)));
+  if (this.settings.trackNamedPages && name) {
+    send(msg.track(name), fn);
   }
 
-  // call track with `msg`.
-  function track(msg){
-    return function(done){
-      self.track(msg, function(err, arr){
-        if (err) return done(err);
-        done(null, arr[1]);
-      });
-    };
+  // call via track method with `msg`.
+  function send(msg, done){
+    self.track(msg, function(err, arr){
+      if (err) return done(err);
+      done(null, arr[1]);
+    });
   }
-
-  batch.end(fn);
```

Comment:
hey guys, we have to call a callback here or else it won't actually complete, and it'll end up hanging.